src/sagemaker/jumpstart/validators.py

```diff
@@ -0,0 +1,153 @@
+            )
+
+        if hyperparameter_spec.type == VariableTypes.INT.value:
+            hyperparameter_value_str = str(hyperparameter_value)
+            start_index = 0
+            if hyperparameter_value_str[0] in ["+", "-"]:
+                start_index = 1
+            if not hyperparameter_value_str[start_index:].isdigit():
+                raise JumpStartHyperparametersError(
+                    f"Hyperparameter '{hyperparameter_name}' must be integer type ('{hyperparameter_value}')."
+                )
+
+        if getattr(hyperparameter_spec, "min", None):
+            if numeric_hyperparam_value < hyperparameter_spec.min:
+                raise JumpStartHyperparametersError(
+                    f"Hyperparameter '{hyperparameter_name}' can be no less than {hyperparameter_spec.min}."
+                )
+
+        if getattr(hyperparameter_spec, "max", None):
+            if numeric_hyperparam_value > hyperparameter_spec.max:
+                raise JumpStartHyperparametersError(
+                    f"Hyperparameter '{hyperparameter_name}' can be no greater than {hyperparameter_spec.max}."
+                )
+
+
+def validate_hyperparameters(
+    model_id: str,
+    model_version: str,
+    hyperparameters: dict,
+    validation_mode: HyperparameterValidationMode = HyperparameterValidationMode.VALIDATE_PROVIDED,
+    region: Optional[str] = JUMPSTART_DEFAULT_REGION_NAME,
+):
+    """Validate hyperparameters for JumpStart models.
+
+    Args:
+        model_id (str): Model ID of the model for which to validate hyperparameters.
+        model_version (str): Version of the model for which to validate hyperparameters.
+        hyperparameters (dict): Hyperparameters to validate.
+        validation_mode (HyperparameterValidationMode): Method of validation to use with
+          hyperparameters. If set to ``VALIDATE_PROVIDED``, only hyperparameters provided
+          to this function will be validated, the missing hyperparameters will be ignored.
+          If set to``VALIDATE_ALGORITHM``, all algorithm hyperparameters will be validated.
+          If set to ``VALIDATE_ALL``, all hyperparameters for the model will be validated.
+        region (str): Region for which to validate hyperparameters. (Default: JumpStart
+          default region).
+
+    """
+
+    if validation_mode is None:
+        validation_mode = HyperparameterValidationMode.VALIDATE_PROVIDED
+
+    if region is None:
+        region = JUMPSTART_DEFAULT_REGION_NAME
+
+    model_specs = jumpstart_accessors.JumpStartModelsAccessor.get_model_specs(
+        region=region, model_id=model_id, version=model_version
+    )
+    hyperparameters_specs = model_specs.hyperparameters
+
+    if validation_mode == HyperparameterValidationMode.VALIDATE_PROVIDED:
+        for hyperparam_name, hyperparam_value in hyperparameters.items():
+            _validate_hyperparameter(hyperparam_name, hyperparam_value, hyperparameters_specs)
```

Comment:
question: this implementation fails fast, ie it raises the first validation exception it encounters. For example, if a model takes hp [`a`, `b`, `c`, `d`], and `a`, `c` and `d` are wrong, it will take the users 3 attempts to fix it. Do you think that is a concern?

Reply:
I think this is fine. Many validators fail at the first error, and force the user to correct upstream errors before tackling issues that occur downstream.